ArduCopter/mode_guided.cpp

```diff
@@ -993,10 +974,15 @@ void ModeGuided::angle_control_run()
     motors->set_desired_spool_state(AP_Motors::DesiredSpoolState::THROTTLE_UNLIMITED);
 
     // call attitude controller
-    if (guided_angle_state.use_yaw_rate) {
-        attitude_control->input_euler_angle_roll_pitch_euler_rate_yaw(roll_in, pitch_in, yaw_rate_in);
+    if (is_zero(guided_angle_state.attitude_quat.length())) {
```

Comment:
Ahh I now see why you constructed a Quat(0,0,0,0). Might be nicer to just pass in attitude_ignore into here instead for those reading later.  

```suggestion
    if (ignore_attitude) {
```

Reply:
We don't have this variable.

Reply:
We don't have this parameter. What we have now works so I would suggest we move forward with this and improve it in another PR.

Reply:
> Yeah ignore_attitude is up in where the type-mask in parsed in GCS_MAVLINK.
I'm ok with not fixing this for now. But lets leave the conversation open so I or someone else sees to come back to this.
I think it is important we check that the user's input quaternion meets our unit length expectation. But that can be done separately.


I think we can only ensure the quaternion is not zero as they are rarely exactly unit length. This is why we renormalise them regularly to ensure they don't move too far from unit length. We currently use the zero length to denote undefined so we have explicitly handled that case making normalisation safe after that.

Reply:
Oh certainly. I didn't mean it had to be exactly unit length, but it should be close coming into the system. If is not "near ||q_vec|| = 1" the math could get weird fast as there are assumptions made that quaternions are near unit length.

Reply:
let's add an is_zero() on Quaternion object